pkg/api/routes.go

```diff
@@ -140,7 +140,7 @@ func (rh *RouteHandler) CheckVersionSupport(response http.ResponseWriter, reques
 	response.Header().Set(constants.DistAPIVersion, "registry/2.0")
 	// NOTE: compatibility workaround - return this header in "allowed-read" mode to allow for clients to
 	// work correctly
-	if rh.c.Config.HTTP.AllowReadAccess {
+	if checkAnonymousPolicyExists(rh.c.Config.AccessControl) {
```

Comment:
I don't know if this makes sense though...
Maybe we should use the request context to mark a user as anonymous?